types/src/v2/params.rs

```diff
@@ -150,12 +150,17 @@ impl<'a> RpcParams<'a> {
 pub struct RpcParamsSequence<'a>(&'a str);
 
 impl<'a> RpcParamsSequence<'a> {
-	fn next_inner<T>(&mut self) -> Option<Result<T, CallError>>
+	fn next_inner<T>(&mut self, treat_empty_json_as_no_params: bool) -> Option<Result<T, CallError>>
 	where
 		T: Deserialize<'a>,
 	{
 		let mut json = self.0.trim_start();
```

Comment:
What do you think of doing the trimming in `RpcParams::new` instead?

Reply:
Yeah that we would much cleaner or in `RpcParams::sequence` but that won't be possible to opt-out from unless we did `RpcParams::sequence_ignore_empty` or something like that.

Reply:
Why would you want to opt out of it? I think we can be opinionated and just enforce no surrounding whitespace.

Reply:
I don't have a good reason for opt-out but kept it for being on the safe-side, remove it?

Reply:
Then let's move it to `new` and enforce it for everyone. :)

Reply:
//cc @maciejhirsz :point_up: regarding that that we ignore valid JSON `{} and []`